float32 upperleft_width_percent
float32 upperleft_height_percent
float32 lowerright_width_percent
float32 lowerright_height_percent
